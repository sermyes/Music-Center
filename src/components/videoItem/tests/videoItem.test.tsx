import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import VideoItem from '../videoItem';
import { fakeVideo } from '../../tests/videos';

describe('VideoItem', () => {
  const video = fakeVideo;

  const onVideoClick = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <VideoItem video={video} onVideoClick={onVideoClick} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  beforeEach(() => {
    render(<VideoItem video={video} onVideoClick={onVideoClick} />);
  });

  it('renders video item', () => {
    const img = screen.getByRole('img')! as HTMLImageElement;

    expect(img.src).toBe(video.snippet.thumbnails.medium.url);
    expect(img.alt).toBe(video.snippet.title);
    expect(screen.getByText(video.snippet.title)).toBeInTheDocument();
  });

  it('calls onVideoClick with video when clicked', async () => {
    const videoItem = screen.getByTitle('video');
    await userEvent.click(videoItem);

    expect(onVideoClick).toHaveBeenCalledTimes(1);
    expect(onVideoClick).toHaveBeenCalledWith(video);
  });
});
