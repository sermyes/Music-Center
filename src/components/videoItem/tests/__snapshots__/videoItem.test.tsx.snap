// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem renders correctly 1`] = `
<div
  className="container"
  data-testid="video"
>
  <div
    className="video"
    onClick={[Function]}
    title="video"
  >
    <img
      alt="title"
      className="thumbnail"
      src="https://image/"
    />
    <div
      className="metadata"
    >
      <p
        className="title"
      >
        title
      </p>
    </div>
  </div>
</div>
`;
